File: backend/stocksTracker/Models/Stock.cs
using Microsoft.EntityFrameworkCore;

namespace stocksTracker.Models
{
    public class Stock
    {
        public int Id { get; set; }

        public string Symbol { get; set; }

        [Precision(18, 2)]
        public decimal CurrentPrice { get; set; }

        [Precision(18, 2)]
        public decimal Change { get; set; }

        [Precision(18, 2)]
        public decimal ChangePercent { get; set; }

        [Precision(18, 2)]
        public decimal HighPrice { get; set; }

        [Precision(18, 2)]
        public decimal LowPrice { get; set; }

        public DateTime LastUpdated { get; set; }
    }
}

File: frontend/src/types/stock.ts
export interface Stock {
  id: number;
  symbol: string;
  companyName: string;
  currentPrice: number;
  change: number;
  changePercent: number;
  highPrice: number;
  lowPrice: number;
  lastUpdated: string;
}
export interface StockSummary {
  totalStocks: number;
  averagePrice: number;
  totalValue: number;
  bestPerformer: string;
  worstPerformer: string;
}
